mono/mini/mini-llvm.c

```diff
@@ -5937,6 +5976,10 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			break;
 		}
 		case OP_GC_SAFE_POINT: {
+			/* LLVM pass PlaceSafepoints will take care of that. See LLVMSetGC */
+			if (cfg->compile_aot && !cfg->llvm_only)
```

Comment:
Why is this disabled for llvmonly? The bitcode we generate should be the output of any passes we add to `mono/llvm`. This isn't in the actual codegen/Xcode-clang step is it? 

Is this optimization going to be enabled on wasm? We're really sensitive to code size there.

Reply:
Because this relies on LLVM having a `MonoGC` which certainely doesn't exist in Apple's LLVM on their end.

Reply:
Wouldn't this lead to a perf loss ? The current code checks sreg1 and only makes the call if its set. The new code will call gc.safepoint_poll which will a c function in  the runtime.

Reply:
@vargaz this is indeed a problem right now, and I won't want this PR to be merged until that's fixed. I just need to figure out my way with emitting the right LLVM IR.

Reply:
@vargaz that is now fixed.